tests/backup/backup_psa_test.go

```diff
@@ -1,30 +1,551 @@
+
+					// If cluster level is restricted and namespace level is baseline or privileged skip the restore as the apps won't come up
+					if psaLevel == "restricted" && (strings.Contains(bkpNamespaces[i], "baseline") || strings.Contains(bkpNamespaces[i], "privileged")) {
+						continue
+					}
+
+					wg.Add(1)
+					go func(backupName string, i int) {
+						defer wg.Done()
+						defaultRestoreName := fmt.Sprintf("%s-%s-custom-ns-sc", RestoreNamePrefix, backupName)
+						pvcs, err := core.Instance().GetPersistentVolumeClaims(bkpNamespaces[i], labelSelectors)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Getting all PVCs from namespace [%s]. Total PVCs - %d", bkpNamespaces[i], len(pvcs.Items)))
+						singlePvc := pvcs.Items[0]
+						sourceScName, err := core.Instance().GetStorageClassForPVC(&singlePvc)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Getting SC from PVC - %s", singlePvc.GetName()))
+						storageClassMapping[sourceScName.Name] = scName
+						namespaceMapping[bkpNamespaces[i]] = bkpNamespaces[i] + "-restored"
+
+						err = CreateRestoreWithReplacePolicyWithValidation(defaultRestoreName, backupName, namespaceMapping, DestinationClusterName, BackupOrgID, ctx, storageClassMapping, ReplacePolicyDelete, scheduledAppContexts)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Create restore with NS and StorageClass mapping with NS level PSA [%s]", defaultRestoreName))
+						restoreNames = append(restoreNames, defaultRestoreName)
+					}(backupName, i)
+				}
+				wg.Wait()
+			})
+
+			Step("Restoring only PVC from backup", func() {
+				ctx, err := backup.GetAdminCtxFromSecret()
+				log.FailOnError(err, "Fetching px-central-admin ctx")
+				log.InfoD("Restoring backed up applications")
+				for i, backupName := range backupNames {
+					restoreName := fmt.Sprintf("%s-%s-%v", RestoreNamePrefix, backupName, time.Now().Unix())
+					namespaceMapping[bkpNamespaces[i]] = bkpNamespaces[i] + "-restored-only-pvc"
+					_, err := CreateCustomRestoreWithPVCs(restoreName, backupName, namespaceMapping, SourceClusterName, BackupOrgID, ctx, make(map[string]string), bkpNamespaces[i])
+					log.FailOnError(err, "Restoring of backup [%s] has failed with name [%s] in namespace [%s]", backupName, restoreName, fmt.Sprintf(bkpNamespaces[i]+"-restored-only-pvc"))
+					restoreNames = append(restoreNames, restoreName)
+				}
+			})
+
+			Step("Revert Cluster Level PSA settings", func() {
+				err := SetDestinationKubeConfig()
+				dash.VerifyFatal(err, nil, "Setting destination kubeconfig")
+				err = RevertClusterLevelPSA()
+				dash.VerifyFatal(err, nil, "Revert cluster level PSA configuration")
+
+				err = SetSourceKubeConfig()
+				dash.VerifyFatal(err, nil, "Setting source kubeconfig")
+				err = RevertClusterLevelPSA()
+				dash.VerifyFatal(err, nil, "Revert cluster level PSA configuration")
+			})
+
+			Step("Taking backup of all the namespaces created with namespace level PSA and no cluster level settings", func() {
```

Comment:
Are the below scenarios not exact same as the initial scenarios
Can we tweak the scenarios little
eg: create new app in new ns instead of using the same ns
Take backup with multiple app with diff psa in one backup   label and restore them

Reply:
Done